cas-mfa-java/src/main/java/net/unicon/cas/mfa/web/support/DefaultMultiFactorAuthenticationSupportingWebApplicationService.java

```diff
@@ -1,121 +1,123 @@
-    }
-
-    @Override
-    public AuthenticationMethodSource getAuthenticationMethodSource() {
-        return this.authenticationMethodSource;
-    }
-}
+package net.unicon.cas.mfa.web.support;
+
+import org.apache.commons.lang.builder.HashCodeBuilder;
+import org.jasig.cas.authentication.principal.AbstractWebApplicationService;
+import org.jasig.cas.authentication.principal.Response;
+import org.jasig.cas.authentication.principal.SimpleWebApplicationServiceImpl;
+import org.jasig.cas.util.HttpClient;
+import org.jasig.cas.authentication.principal.Response.ResponseType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.validation.constraints.NotNull;
+
+/**
+ * The MultiFactorAuthenticationService is an extension of the generic CAS service
+ * that delegates calls to {@link SimpleWebApplicationServiceImpl}. The only difference
+ * is that it is only is activated when the request parameter {@link #CONST_PARAM_AUTHN_METHOD} is
+ * present and its value is supported by the corresponding argument extractor.
+ * <p>NOTE: The delegation is necessary because the {@link SimpleWebApplicationServiceImpl}
+ * itself is marked as final. Future versions of CAS might make the class more available.
+ *
+ * @author Misagh Moayyed
+ */
+public final class DefaultMultiFactorAuthenticationSupportingWebApplicationService
+        extends AbstractWebApplicationService
+        implements MultiFactorAuthenticationSupportingWebApplicationService {
+
+    /** The logger instance. **/
+    protected static final Logger LOGGER =
+            LoggerFactory.getLogger(DefaultMultiFactorAuthenticationSupportingWebApplicationService.class);
+
+    private static final long serialVersionUID = 7537062414761087535L;
+
+    /** The wrapped service. */
+    private final SimpleWebApplicationServiceImpl wrapperService;
+
+    /** The authentication method. */
+    private final String authenticationMethod;
+
+    /** The authentication method source. */
+    private AuthenticationMethodSource authenticationMethodSource;
+
+    /**
+     * Create an instance of {@link DefaultMultiFactorAuthenticationSupportingWebApplicationService}.
+     *
+     * @param id the service id, potentially with a jsessionid; still needing excised
+     * @param originalUrl the service url
+     * @param artifactId the artifact id
+     * @param responseType the HTTP method for the response
+     * @param httpClient http client to process requests
+     * @param authnMethod the authentication method required for this service
+     */
+    public DefaultMultiFactorAuthenticationSupportingWebApplicationService(final String id, final String originalUrl,
+            final String artifactId, final ResponseType responseType, final HttpClient httpClient, @NotNull final String authnMethod) {
+        super(cleanupUrl(id), originalUrl, artifactId, httpClient);
+        this.wrapperService = new SimpleWebApplicationServiceImpl(id, originalUrl, artifactId, responseType, httpClient);
```

Comment:
How do we pass the `ResponseType` to `this.wrapperService`? Do we not need to in order for this to work?